avocado/utils/asset.py

```diff
@@ -151,20 +151,33 @@ def _get_hash_from_file(self, asset_path):
         :returns: the hash, if it exists.
         :rtype: str
         """
-        discovered = None
         hash_file = self._get_hash_file(asset_path)
         if not os.path.isfile(hash_file):
             self._create_hash_file(asset_path)
 
-        with open(hash_file, 'r') as hash_file:
-            for line in hash_file:
-                # md5 is 32 chars big and sha512 is 128 chars big.
-                # others supported algorithms are between those.
-                pattern = '%s [a-f0-9]{32,128}' % self.algorithm
-                if re.match(pattern, line):
-                    discovered = line.split()[1]
-                    break
-        return discovered
+        return Asset.read_hash_from_file(hash_file)[1]
```

Comment:
I converted this piece of code into a classmethod because I didn't want to force developers to instantiate a new object just to read a hash from a file. 

Yes, we could use here a call with `self`, and I think it is just a matter of style. Using the class, creates a standard all over the code in all calls, since this could be called inside and outside objects. Not sure if we have (or will have) examples in this code of subclasses of `Asset` to worry about the override.

We could think about the drawbacks of calling in the class if you don't like this style.

Reply:
The only issue is the MRO. IMO, unless there's a reason for pinning the MRO, we should follow general OO expectations, as this ends up with the least amount of surprise for users.

Reply:
Where do you see a problem with MRO? Not sure if I got it. Today I could not find any class that inherits from Asset.